### PR TITLE
=str deprecate ActorPublisher/Subscriber, use GraphStage

### PR DESCRIPTION
**Pull request by ktoso**
Refs https://github.com/akka/akka/issues/21423

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/actor/ActorPublisher.scala [hunk 1]
```diff
@@ -12,6 +12,8 @@ import concurrent.duration.FiniteDuration
 import akka.stream.impl.CancelledSubscription
 import akka.stream.impl.ReactiveStreamsCompliance._
 
+@Deprecated
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/actor/ActorPublisher.scala]: we have not used this annotation previously?

**Review by patriknw — APPROVED** [review]
LGTM, with question about the annotation

File: akka-docs/rst/java/stream/stream-integrations.rst [hunk 2]
```diff
@@ -437,6 +437,14 @@ These can be consumed by other Reactive Stream libraries or used as an Akka Stre
 ActorPublisher
 --------------
 
+.. warning::
+  **Deprecation warning:** ``ActorPublisher`` is deprecated in favour of the vastly more 
+  type-safe and safe to implement :class:`akka.stream.stage.GraphStage`. It may also
```
Comment (drewhk) [inline akka-docs/rst/java/stream/stream-integrations.rst]: it may -> it can?

**Review by drewhk — DISMISSED** [review]
I think the docs on the GraphStages needs to be expanded, and the "experimental" note removed. These changes should also be backported to 2.4 (except the deprecation ofc)

File: akka-docs/rst/java/stream/stream-integrations.rst (hunk 2 above)
Comment (ktoso) [inline akka-docs/rst/java/stream/stream-integrations.rst]: ok

**Comment by ktoso** [comment]
I noticed we also should deprecate Source.actorPublisher etc

**Comment by ktoso** [comment]
I added more notes, please have a look @akka/akka-team

**Review by johanandren — APPROVED** [review]
LGTM!

**Pull request merged**